e2e/testcases/reconciler_manager_test.go

```diff
@@ -603,3 +603,37 @@ func TestAutopilotReconcilerAdjustment(t *testing.T) {
 	})
 
 }
+
+func TestAllowVerticalScaleResourceAdjustment(t *testing.T) {
+	nt := nomostest.New(t, ntopts.SkipMonoRepo)
+
+	reconcilerDeployment := &appsv1.Deployment{}
+	if err := nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem, reconcilerDeployment); err != nil {
+		nt.T.Fatal(err)
+	}
+	rs := &v1beta1.RootSync{}
+	if err := nt.Get(configsync.RootSyncName, configsync.ControllerNamespace, rs); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	_, err := nt.Kubectl("patch", "deployment", "reconciler-manager", "-n", "config-management-system", "-p", "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"args\":[\"--enable-leader-election\", \"--allow-vertical-scale\"],\"name\":\"reconciler-manager\"}]}}}}")
+	if err != nil {
+		nt.T.Fatalf("Error enabling vertical scale in deployment", err)
+	}
+
+	var updatedFirstContainerCPURequest int64
+	var updatedFirstContainerMemoryRequest int64
+	firstContainerName := reconcilerDeployment.Spec.Template.Spec.Containers[0].Name
+	updatedFirstContainerCPURequest = 159
+	updatedFirstContainerMemoryRequest = 259
+
+	nt.MustMergePatch(rs, fmt.Sprintf(`{"spec":{"override":{"resources":[{"containerName":"%s","memoryRequest":"%dMi", "cpuRequest":"%dm"}]}}}`,
+		firstContainerName, updatedFirstContainerMemoryRequest, updatedFirstContainerCPURequest))
+
+	// generation = 2 because the applied changes will not be reverted
+	nomostest.Wait(nt.T, "Waiting for fourth reconciliation", nt.DefaultWaitTimeout, func() error {
```

Comment:
Why is this the fourth reconciliation?

Reply:
Updated logs to be more clear

Reply:
I didn't see your update. Did you push the change?

Reply:
sorry, looks like i lost a few commits when squashing them. Fixing now